metrics/src/datapoint.rs

```diff
@@ -88,43 +88,56 @@ macro_rules! datapoint {
     };
 }
 
+#[macro_export]
+macro_rules! datapoint_log {
```

Comment:
Why didn't you just add this to `datapoint!`?

Reply:
because I want this wrapping the calls to submit before the point creation.

Reply:
You can do that within `datapoint!` and avoid creating a new thing

Reply:
I don't see how, `datapoint!` doesn't take a log level in all its clauses.  We need the filter to run before the constructor is called, otherwise we are just wasting cycles.

it prolly makes sense to remove `if log::log_enabled!(log::Level::Debug) {` from `datapoint!`

Reply:
> that wont prevent the constructors from running prior to run. If you want the data, it should be info and produced no more than once per slot. or use a counter which emit once per 1k samples.

So the argument is by dropping all datapoints < info by default, you're forced to use `datapoint_info!`.  In doing so you'll spam the log if your datapoint is too chatty and somebody will eventually notice and yell at you.   🤷‍♂, I'm ok with that approach for now

Reply:
How about we rename  `datapoint!` to `_interrnal_datapoint!` and not export it, then rename `datapoint_log` to `datapoint`, if you don't want to merge the two.

Reply:
Ah, how about this instead:
* `datapoint!` -> `create_datapoint!`
* `datapoint_log!` -> `datapoint!`